errorHandling.java

```diff
@@ -0,0 +1,52 @@
+package sample;
+
+public class ErrorHandling {
+
+  
+  public Object createUser(User user){
+      
+      if(loadUser(user.getId()) instanceof User){
+        // user already exists
+        return USER_ALREADY_EXISTS; // error flag
+        return 101; // or error code
+      };
+      
+      resolveGroup(user);
+      return userRepository.save(user);
+  }
+  
```

Comment:
// 2. Use Try-Catch-Finally Statements
  // Good
  public void handleSimulation(Simulation simulation){
    try{
      resultHandler.handle(simulation);
    }
    catch (Throwable t){
      log.error("failed to save result", t);
    }
    finally {
      simulationService.remove(simulation); // with finally we try to have consistent state if something goes wrong
    }
  }